Framework/Algorithms/test/CorelliPowderCalibrationDatabaseTest.h

```diff
@@ -0,0 +1,431 @@
+  }
+  static void destroySuite(CorelliPowderCalibrationDatabaseTest *suite) {
+    delete suite;
+  }
+
+  //-----------------------------------------------------------------------------
+  void test_Init() {
+    CorelliPowderCalibrationDatabase alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+  }
+
+  //-----------------------------------------------------------------------------
+  /**
+   * @brief Test basic file IO library methods
+   */
+  void test_file_io() {
+    // create directory
+    std::string test_dir{"TestCorelliPowderCalibrationX"};
+    boost::filesystem::create_directory(test_dir);
+    TS_ASSERT(boost::filesystem::is_directory(test_dir));
+
+    // clean
+    boost::filesystem::remove_all(test_dir);
+  }
+
+  //-----------------------------------------------------------------------------
+  /**
+   * @brief Test algorithm to convert datetime string to date stamp
+   */
+  void test_timestamp_conversion() {
+    std::string yyyymmdd = CorelliPowderCalibrationDatabase::convertTimeStamp(
+        "2018-02-20T12:57:17");
+    TS_ASSERT_EQUALS(yyyymmdd, "20180220");
+  }
+
+  //-----------------------------------------------------------------------------
+  /**
+   * @brief Test ComponentPosition
+   */
+  void test_component() {
+    // compare:
+    CorelliCalibration::ComponentPosition pos1{0., 0., 0., 20., 30., 40., 50};
+    CorelliCalibration::ComponentPosition pos2{0., 0., 0., 20., 30., 40., 50};
+    CorelliCalibration::ComponentPosition pos3{0.,  0.,  0., 20.003,
+                                               30., 40., 50};
+    TS_ASSERT(pos1.equalTo(pos2, 1E-7));
+    TS_ASSERT(!pos1.equalTo(pos3, 1E-7));
+  }
+
+  //-----------------------------------------------------------------------------
+  /**
+   * @brief Test CalibrationWworkspaceHandler
+   */
+  void test_calibration_workspace_handler() {
+    // Create a correct calibration worksapce
+    std::string outwsname("CorelliPowderCalibrationDatabaseTest_TableWS2");
+    TableWorkspace_sptr calib_ws =
+        createTestCalibrationTableWorkspace(outwsname);
+
+    // Create an incorrect calibration workspace
+    std::string wrongwsname{
+        "CorelliPowderCalibrationDatabaseTest_TableWS_Wrong"};
+    TableWorkspace_sptr calib_wrong_ws =
+        createIncorrectTestCalibrationTableWorkspace(wrongwsname);
+
+    // Init CalibrationTableHandler instance
+    CorelliCalibration::CalibrationTableHandler calib_handler =
+        CorelliCalibration::CalibrationTableHandler();
+
+    // Expect to fail set a wrong
+    TS_ASSERT_THROWS_ANYTHING(
+        calib_handler.setCalibrationTable(calib_wrong_ws));
+    // Shall not throw
+    TS_ASSERT_THROWS_NOTHING(calib_handler.setCalibrationTable(calib_ws));
+
+    // Test method to retrieve components names (rows)
+    std::vector<std::string> componentnames = calib_handler.getComponentNames();
+    std::vector<std::string> expectednames{"source", "sample", "bank1"};
```

Comment:
to be consistent, can you use CORELLI's component names for source and sample ("moderator" and "sample-position")?